backends/mediatek/runtime/NeuronBackend.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+* Copyright (c) 2024 MediaTek Inc.
+*
+* Licensed under the BSD License (the "License"); you may not use this file
+* except in compliance with the License. See the license file in the root
+* directory of this source tree for more details.
+*/
+
+
+#include "NeuronBackend.h"
+#include "NeuronBufferAllocator.h"
+#include "NeuronPayloadHeader.h"
+#include "NeuronLog.h"
+#include "api/NeuronAdapter.h"
+
+#include "executorch/runtime/core/error.h"
+
+#include <algorithm>
+#include <memory>
+#include <new>
+#include <unordered_set>
+
+
+namespace torch {
+namespace executor {
+
+const char kHighAddrKey[] = "HighAddr";
+const char kImportForeverKey[] = "ImportForever";
+
+Result<DelegateHandle*> NeuronBackend::init(BackendInitContext& context,
+                                            FreeableBuffer* processed,
+                                            ArrayRef<CompileSpec> compile_specs) const {
+    NeuronDelegateSetting setting;
+    for (auto& compile_spec : compile_specs) {
+        if (std::strcmp(compile_spec.key, kHighAddrKey) == 0) {
+            setting.mHighAddr = *static_cast<char*>(compile_spec.value.buffer);
+            LogInfo("NeuronBackend", "IsHighAddr Enable : %d", setting.mHighAddr);
+        } else if (std::strcmp(compile_spec.key, kImportForeverKey) == 0) {
+            setting.mImportForever = *static_cast<char*>(compile_spec.value.buffer);
+            LogInfo("NeuronBackend", "IsImportForever Enable : %d", setting.mImportForever);
+        } else {
+            LogWarn("NeuronBackend", "unknown compile spec: %s", compile_spec.key);
+        }
+    }
+    auto Payload = NeuronPayload(processed->data(), processed->size());
+    LogInfo("NeuronBackend", "version %u, input %u, output %u, length %u, payload size: %zu",
+                 Payload.Header.Version, Payload.Header.InputCount, Payload.Header.OutputCount, Payload.Header.DataLen, processed->size());
+
+    auto delegate = std::unique_ptr<NeuronExecuTorchDelegate>(
+        new (std::nothrow) NeuronExecuTorchDelegate());
+    if (delegate == nullptr) {
+        return nullptr;
+    }
+    auto res = delegate->LoadCompiledNetwork(Payload, setting);
+    return res == NEURON_NO_ERROR ? delegate.release() : nullptr;
+}
+
+Error NeuronBackend::execute(__ET_UNUSED BackendExecutionContext& context,
+                             DelegateHandle* handle,
+                             EValue** args) const {
+    NeuronExecuTorchDelegate* delegate = reinterpret_cast<NeuronExecuTorchDelegate*>(handle);
+    return delegate->execute(context, args);
+}
+
+void NeuronBackend::destroy(DelegateHandle* handle) const {
+    NeuronExecuTorchDelegate* delegate = reinterpret_cast<NeuronExecuTorchDelegate*>(handle);
+    delete delegate;
```

Comment:
you might have to call the destructor directly. Something like this. `delegate->~NeuronExecuTorchDelegate()`